File: 2006-find-the-student-that-will-replace-the-chalk/2006-find-the-student-that-will-replace-the-chalk.cpp
class Solution {
public:
    int chalkReplacer(vector<int>& chalk, int k) {
        ios_base::sync_with_stdio(false);cin.tie(nullptr);cout.tie(nullptr);
        long long int sum = 0, n = chalk.size();
        for (int i=0; i<n; i++) {
            sum += chalk[i];
        }
        k = k % sum;
        for (int i=0; i<n; i++) {
            if (k < chalk[i]) return i;
            k -= chalk[i];
        }
        return -1;
    }
};

File: 2006-find-the-student-that-will-replace-the-chalk/2006-find-the-student-that-will-replace-the-chalk.py
class Solution:
    def chalkReplacer(self, chalk: List[int], k: int) -> int:
        total_sum = sum(chalk)
        
        k = k % total_sum
        for i in range(0, len(chalk)):
            if k < chalk[i]:
                return i
            k -= chalk[i]
        
        return -1